src/software/ai/hl/stp/strategy/strategy_impl.hpp

```diff
@@ -194,15 +195,14 @@ Pass StrategyImpl<PitchDivision, ZoneEnum>::getBestCommittedPass()
 template <class PitchDivision, class ZoneEnum>
 std::optional<Shot> StrategyImpl<PitchDivision, ZoneEnum>::getBestShot(const Robot& robot)
 {
-    if (robot_to_best_shot_.contains(robot.id()))
+    if (!best_shot_)
```

Comment:
seeing this pattern quite a bit for checking for best_shot existence - maybe we can fall back on the second or third best for these cases, relating to the suggesting of sorting the sample passes.

Reply:
If best_shot is std::nullopt, that implies that no feasible shot exists (so there isn't a 2nd or 3rd possible shot to fall back on). e.g. maybe we are behind the net